stdlib/domain.ml

```diff
@@ -114,7 +114,7 @@ let self () = Raw.self ()
 
 module DLS = struct
 
-  type 'a key = int ref
+  type 'a key = int ref * (unit -> 'a)
 
   type entry = {key: int ref; slot: Obj.t ref}
```

Comment:
This is a little confusing to me now that we've made `'a key` an `int ref * (unit -> 'a)`
Might it be better to call the `int ref` that is part of the `entry` record the `key_id` or `id`?